### PR TITLE
Compatibility with Julia 1.0

### PR DESCRIPTION
**Pull request by YingboMa**

### CODE REVIEW

File: src/mathfuns.jl [hunk 1]
```diff
@@ -52,8 +54,9 @@ for (meth, libnm) in [
                       (:sqrt,:sqrt),
                       (:exp,:exp),
                       ]
-    eval(Expr(:import, :Base, meth))
-    IMPLEMENT_ONE_ARG_FUNC(:(Base.$meth), libnm)
+    modu = meth != :gamma ? :Base : :SpecialFunctions
```
Comment (isuruf) [inline src/mathfuns.jl]: Can you make this a part of the list above? i.e. `(:abs, :abs, :Base)`? Also can you move, `eta, zeta`, from the list below to the above list?

**Comment by isuruf** [comment]
Thanks @YingboMa for the PR. I'm not sure about adding a new binary dependency `SpecialFunctions.jl` just for the ability to extend it. Is there a way to avoid it?

**Comment by YingboMa** [comment]
I am not sure if there is a way to avoid it. Packages like `ForwardDiff.jl` require `SpecialFunctions` too, so the binary dependency might not be a problem. @isuruf

**Comment by jleugeri** [comment]
Offtopic remark: Since you're already on it, there's also a warning about a missing Libdl dependency on a fresh julia 1.0 with the latest master. No idea how relevant that is, so I didn't want to open a new issue just for this.

> ┌ Warning: Package SymEngine does not have Libdl in its dependencies:
> │ - If you have SymEngine checked out for development and have
> │   added Libdl as a dependency but haven't updated your primary
> │   environment's manifest file, try `Pkg.resolve()`.
> │ - Otherwise you may need to report an issue with SymEngine
> └ Loading Libdl into SymEngine from project dependency, future warnings for SymEngine are suppressed.

**Comment by isuruf** [comment]
LIbdl is a standard library and there's no need to install it. I'm not sure why the warning comes up.

**Comment by jleugeri** [comment]
I **think** it's because whether it's installed or not (it is installed by default), it still needs to be explicitly added to the project dependencies in `Project.toml` for the new package manager?

**Pull request merged**